### PR TITLE
Fixes #35336 - System purpose card should not show when host is unregistered

### PR DESCRIPTION
**Pull request by jeremylenz**
#### What are the changes introduced in this pull request?

The system purpose card was trying to render even for unregistered hosts. This breaks the entire host details page.

This change will check if the host is registered before rendering the system purpose card.

#### What are the testing steps for this pull request?

Check that the new host detail page renders properly with both registered and unregistered hosts.

### CODE REVIEW

**Comment by jeremylenz** [comment]
Updated to also hide the HW Properties card for unregistered hosts

File: webpack/components/extensions/HostDetails/DetailsTabCards/HwPropertiesCard.js [hunk 1]
```diff
@@ -32,6 +33,7 @@ HostDisks.defaultProps = {
 };
 
 const HwPropertiesCard = ({ isExpandedGlobal, hostDetails }) => {
+  if (hostIsNotRegistered({ hostDetails })) return null;
```
Comment (lfu) [inline webpack/components/extensions/HostDetails/DetailsTabCards/HwPropertiesCard.js:36]: 👍🏻

File: webpack/components/extensions/HostDetails/Cards/SystemPurposeCard/SystemPurposeCard.js [hunk 2]
```diff
@@ -57,6 +57,8 @@ const SystemPurposeCard = ({ hostDetails }) => {
     );
   }
 
+  if (hostIsNotRegistered({ hostDetails })) return null;
```
Comment (lfu) [inline webpack/components/extensions/HostDetails/Cards/SystemPurposeCard/SystemPurposeCard.js:60]: Why not check it at the beginning of the function?

Comment (jeremylenz) [reply webpack/components/extensions/HostDetails/Cards/SystemPurposeCard/SystemPurposeCard.js]: @lfu updated :+1:

Comment (jeremylenz) [reply webpack/components/extensions/HostDetails/Cards/SystemPurposeCard/SystemPurposeCard.js]: I guess this is why:
```
Error:   48:29  error  React Hook "useSelector" is called conditionally. React Hooks must be called in the exact same order in every component render  react-hooks/rules-of-hooks
Error:   51:33  error  React Hook "useState" is called conditionally. React Hooks must be called in the exact same order in every component render     react-hooks/rules-of-hooks
```

**Comment by jeremylenz** [comment]
Tried moving that check to the beginning, but that would mean calling hooks conditionally. So I changed it back.

File: webpack/components/extensions/HostDetails/Cards/SystemPurposeCard/SystemPurposeCard.js (hunk 2 above)
Comment (lfu) [reply webpack/components/extensions/HostDetails/Cards/SystemPurposeCard/SystemPurposeCard.js:60]: We can move them together.
```
const SystemPurposeCard = ({ hostDetails }) => {
  const hostDetailsStatus = useSelector(selectHostDetailsStatus);
  const [editing, setEditing] = useState(false);
  if (hostIsNotRegistered({ hostDetails })) return null;
```

Comment (jeremylenz) [reply webpack/components/extensions/HostDetails/Cards/SystemPurposeCard/SystemPurposeCard.js:60]: But then we would have logic in the middle of assigning all of the consts. Seems cleaner to just assign all the consts, and then have your `if`s..

**Review by lfu — APPROVED** [review]
LGTM 👍🏻 

APJ

**Comment by jeremylenz** [comment]
[test katello]

**Pull request merged**